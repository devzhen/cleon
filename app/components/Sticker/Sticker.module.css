.container {
  position: absolute;
  width: 178px;
  height: 181px;
  padding: 8px; 
  background-color: var(--color-modal-background);
  border: 1px solid white;
  box-shadow: 0 4px 18px rgba(0, 0, 0, 0.18);
  cursor: pointer;
  user-select: none;
}

.container span[data-role="date"] {
  color: var(--color-date-color);
  font-size: 10px;
  margin-left: 7px;
  margin-bottom: 5px;
}

.container div[data-role="content-wrapper"] {
  height: 124px;
  border: var(--color-text-area-border);
  overflow: hidden;
  outline: none;
}

.container div[data-role="content-wrapper"] span {
  color: var(--color-text-area-color);
  font-size: 13px;
  margin-top: 10px;
  margin-left: 10px;
  margin-right: 10px;
  line-height: 1.5;
}

.container div[data-role="image-wrapper"] {
  display: flex;
  justify-content: flex-end;
}

.container img {
  width: 13px;
  height: 13px;
  margin-top: 10px;
  cursor: pointer;
}