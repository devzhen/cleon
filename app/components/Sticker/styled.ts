import styled from "styled-components";

export const Container = styled.div<{
  $top?: number;
  $left?: number;
  $zIndex?: number;
}>`
  position: absolute;
  width: 178px;
  height: 181px;
  padding: 8px; 
  background-color: #fff8c8;
  border: 1px solid white;
  box-shadow: 0 4px 18px rgba(0, 0, 0, 0.18);
  z-index: 2;
  cursor: pointer;
  user-select: none;

  ${(props) => props.$top && `top: ${props.$top}px`};
  ${(props) => props.$left && `left: ${props.$left}px`};
  ${(props) => props.$zIndex && `z-index: ${props.$zIndex}`};
`;

export const Date = styled.div`
  color: #979797;
  font-size: 10px;
  margin-left: 7px;
  margin-bottom: 5px;
`;

export const ContentWrapper = styled.div`
  height: 124px;
  border: 1px solid #dfe2e7;
  overflow: hidden;
  outline: none;
`;

export const Content = styled.span`
  color: #2e2e2e;
  font-size: 13px;
  margin-top: 10px;
  margin-left: 10px;
  margin-right: 10px;
  line-height: 1.5;
`;

export const ImageWrapper = styled.div`
  display: flex;
  justify-content: flex-end;

  img {
    width: 13px;
    height: 13px;
    margin-top: 10px;
    cursor: pointer;
  }
`;
