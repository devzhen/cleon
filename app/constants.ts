export const DEFAULT_COORDS = {
  top: 100,
  left: 220,
} as const;

export const BOARD_TYPE = {
  left: 'left',
  right: 'right',
} as const;