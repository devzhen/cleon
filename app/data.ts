import { v4 as uuid } from "uuid";

export const stickers = [
  {
    id: uuid(),
    top: 110,
    left: 230,
    createdAt: new Date().toISOString(),
    text: ''
  }
];
