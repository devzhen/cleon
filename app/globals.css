:root {
  --color-active-board: #eb6529;
  --color-not-active-board: #bfc3ca;
  --color-main-background: #f0cc46;
  --color-board-background: #e9ecf1;
  --color-header-right-side: #7b7b7b;
  --color-modal-background: #fff8c8;
  --color-text-area-background: #fffde6;
  --color-text-area-border: #dfe2e7;
  --color-text-area-color: #2e2e2e;
  --color-date-color: #979797;
}

html,
body {
  max-width: 100vw;
}

* {
  box-sizing: border-box;
  padding: 0;
  margin: 0;
}

.main {
  display: grid;
  grid-template-rows: 72px 1fr;
  grid-template-columns: 638px 1fr;
  height: 100dvh;
  width: 100%;
}

.boardActive {
  border-color: var(--color-active-board) !important;
}

.boardNotActive {
  border-color: var(--color-not-active-board) !important;
}
