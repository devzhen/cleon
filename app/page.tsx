"use client";

import styled from "styled-components";

import Header from "./components/Header";
import LeftSide from "./components/LeftSide";

export default function Home() {
  return (
    <Main>
      <header>
        <Header />
      </header>
      <LeftSide />
    </Main>
  );
}

const Main = styled.main`
  display: grid;
  grid-template-rows: 72px 1fr;
  grid-template-columns: 638px 1fr;
  height: 100dvh;
  width: 100%;

  header {
    grid-column: 2 span;
    height: 66px;
    width: 100%;
  }
`;
