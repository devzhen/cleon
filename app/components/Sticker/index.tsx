import { parseISO, format } from "date-fns";
import Image from "next/image";

import { Sticker } from "@/app/data";

import {
  Container,
  Date,
  Content,
  ContentWrapper,
  ImageWrapper,
} from "./styled";

type StickerProps = {
  sticker: Sticker;
  onMouseDown: VoidFunction;
  // eslint-disable-next-line unused-imports/no-unused-vars
  editSticker: (e: MouseEvent) => void;
};

const Sticker = (props: StickerProps) => {
  const { sticker, onMouseDown, editSticker } = props;

  const date = parseISO(sticker.createdAt);
  const dateFormatted = format(date, "yyyy LLLL d, HH:mm:ss");

  return (
    <Container
      onMouseDown={onMouseDown}
      $top={sticker.top}
      $left={sticker.left}
      className="sticker"
    >
      <Date>{dateFormatted}</Date>
      <ContentWrapper>
        <Content>{sticker.text}</Content>
      </ContentWrapper>
      <ImageWrapper>
        <Image
          src="/reg.png"
          width={13}
          height={13}
          alt="Picture of the author"
          draggable={false}
          onClick={editSticker as VoidFunction}
        />
      </ImageWrapper>
    </Container>
  );
};

export default Sticker;
