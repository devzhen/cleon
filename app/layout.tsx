import type { Metadata } from "next";
import { Roboto } from "next/font/google";

import "./globals.css";

import StyledComponentsRegistry from "../lib/registry";

const roboto = Roboto({
  weight: ["400", "700"],
  style: ["normal"],
  subsets: ["latin"],
  display: "swap",
});

export const metadata: Metadata = {
  title: "Cleon",
  description: "Cleon",
};

export default function RootLayout({
  children,
}: Readonly<{
  children: React.ReactNode;
}>) {
  return (
    <html lang="en">
      <body className={roboto.className}>
        <StyledComponentsRegistry>{children}</StyledComponentsRegistry>
      </body>
    </html>
  );
}
