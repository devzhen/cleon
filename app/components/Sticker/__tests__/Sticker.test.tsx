import { render } from '@testing-library/react';

import { BOARD_TYPE } from '@/app/constants';
import { firstId, stickers } from '@/app/data';

import Sticker from '..';

describe('Sticker tests', () => {
  const props = {
    board: BOARD_TYPE.left,
    closeModal: jest.fn(),
    createSticker: jest.fn(),
    sticker: stickers[firstId],
    editSticker: jest.fn(),
    isDeleteMode: false,
    onMouseDown: jest.fn(),
    removeSticker: jest.fn(),
  };
  it('matches the snapshot', () => {
    const { container } = render(<Sticker {...props} />);

    expect(container).toMatchSnapshot();
  });

  it('matches the snapshot when isDeleteMode is true', () => {
    const newProps = {
      ...props,
      isDeleteMode: true,
    };

    const { container } = render(<Sticker {...newProps} />);

    expect(container).toMatchSnapshot();
  });
});
