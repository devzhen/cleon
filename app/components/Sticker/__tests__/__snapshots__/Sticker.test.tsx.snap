// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sticker tests matches the snapshot 1`] = `
<div>
  <div
    class="container sticker"
    data-board="left"
    data-id="uuid/v4"
    role="presentation"
    style="top: 449px; left: 963px; z-index: 3;"
  >
    <span
      data-role="date"
    >
      02-03-2024 09:10:11 AM
    </span>
    <div
      data-role="content-wrapper"
    >
      <span>
        Draggable element 4
      </span>
    </div>
    <div
      data-role="image-wrapper"
    >
      <img
        alt="Picture of the author"
        data-nimg="1"
        decoding="async"
        draggable="false"
        height="13"
        loading="lazy"
        src="/_next/image?url=%2Freg.png&w=32&q=75"
        srcset="/_next/image?url=%2Freg.png&w=16&q=75 1x, /_next/image?url=%2Freg.png&w=32&q=75 2x"
        style="color: transparent;"
        width="13"
      />
    </div>
  </div>
</div>
`;

exports[`Sticker tests matches the snapshot when isDeleteMode is true 1`] = `
<div>
  <div
    class="container sticker"
    data-board="left"
    data-id="uuid/v4"
    role="presentation"
    style="top: 449px; left: 963px; z-index: 3;"
  >
    <span
      data-role="date"
    >
      02-03-2024 09:10:11 AM
    </span>
    <div
      data-role="content-wrapper"
    >
      <span>
        Draggable element 4
      </span>
    </div>
    <div
      data-role="image-wrapper"
    >
      <img
        alt="Picture of the author"
        data-nimg="1"
        decoding="async"
        draggable="false"
        height="13"
        loading="lazy"
        src="/_next/image?url=%2Fdel.png&w=32&q=75"
        srcset="/_next/image?url=%2Fdel.png&w=16&q=75 1x, /_next/image?url=%2Fdel.png&w=32&q=75 2x"
        style="color: transparent;"
        width="13"
      />
    </div>
  </div>
</div>
`;
